### PR TITLE
[FLINK-30062][Connectors/HBase] Externalize the existing connector code as is

### PR DESCRIPTION
**Pull request by ferenc-csaky**
I moved the business code as is. Some changes were made:
* Maven dependency structure. I tried to move common things under `dependencyManagement` to use the same version.
* I added a default `2.8.5` Hadoop version in both `ITCase` setup validation checks to trigger the test, cause in its current format it checks an environment variable, which is not set locally.

### CODE REVIEW

**Comment by MartijnVisser** [comment]
@ferenc-csaky Not sure if you noticed but the build failed

**Comment by ferenc-csaky** [comment]
> @ferenc-csaky Not sure if you noticed but the build failed

Yep, I did, I'm working on the dependency convergence.

**Comment by ferenc-csaky** [comment]
Frankly, achieving dependency convergence made dependency management and exclusions quite messy at this point and the main cause of that is some Hadoop dependencies are the same between base, 1.4 and 2.2, while the HBase related things differ.

As we move forward with dropping the HBase 1.x functionality I expect this to become a lot more cleaner, but I do not want to make too many changes in 1 step.

**Comment by zentol** [comment]
FYI, we'd like to retain the git history. The [externalization guide](https://cwiki.apache.org/confluence/display/FLINK/Externalized+Connector+development) contains some hints to that end.

**Comment by zentol** [comment]
I'm not surprised that dependency convergence is a pain. It is one of the big reasons why we haven't done it in Flink already.

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+		<url>https://github.com/apache/flink-connector-hbase</url>
+		<connection>git@github.com:apache/flink-connector-hbase.git</connection>
+		<developerConnection>scm:git:https://gitbox.apache.org/repos/asf/flink-connector-hbase.git</developerConnection>
+	</scm>
+
+	<properties>
+		<scala.binary.version>2.12</scala.binary.version>
+
+		<flink.version>1.16.0</flink.version>
+		<flink.shaded.version>16.0</flink.shaded.version>
```
Comment (zentol) [inline pom.xml]:
```suggestion
		<flink.shaded.version>15.0</flink.shaded.version>
```
This is the actual version that 1.16.0 depends on.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+		<hbase1.version>1.4.3</hbase1.version>
+		<hbase2.version>2.2.3</hbase2.version>
+		<httpclient.version>4.5.13</httpclient.version>
+		<httpcore.version>4.4.14</httpcore.version>
+		<jackson.version>2.13.4.20221013</jackson.version>
+		<jsr305.version>1.3.9</jsr305.version>
+		<junit4.version>4.13.2</junit4.version>
+		<junit5.version>5.8.1</junit5.version>
+		<kryo.verison>2.24.0</kryo.verison>
+		<netty.version>4.1.77.Final</netty.version>
```
Comment (zentol) [inline pom.xml]: I'd suggest to stick to the version that we used in 1.16.0. We've had issues with netty versions way too often that I'd be comfortable with just bumping it just like that.

File: flink-connector-hbase-2.2/pom.xml [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+		</dependency>
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-hadoop-compatibility_${scala.binary.version}</artifactId>
+			<scope>test</scope>
+			<exclusions>
+				<exclusion>
+					<groupId>org.apache.flink</groupId>
+					<artifactId>flink-shaded-include-yarn_${scala.binary.version}</artifactId>
```
Comment (zentol) [inline flink-connector-hbase-2.2/pom.xml]: There's no such dependency.

File: flink-connector-hbase-2.2/pom.xml [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-table-common</artifactId>
+			<type>test-jar</type>
+			<scope>test</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-table-api-scala-bridge_${scala.binary.version}</artifactId>
```
Comment (zentol) [inline flink-connector-hbase-2.2/pom.xml]: what do we need the scala bridge for? I see no scala code or any scala import.

File: .idea/vcs.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="" vcs="Git" />
```
Comment (zentol) [inline .idea/vcs.xml:22]: This should be closer to https://github.com/apache/flink-connector-elasticsearch/blob/main/.idea/vcs.xml for some nifty IDE features.

**Comment by ferenc-csaky** [comment]
Messed up the rebase first time, so this one got closed cause the init commit was on the top. That is fixed now and the comments added also addressed, pls. check #2.

File: flink-connector-hbase-2.2/pom.xml (hunk 4 above)
Comment (ferenc-csaky) [reply flink-connector-hbase-2.2/pom.xml]: It was this way already, I just did not touch it, but you're right, I removed it.

**Pull request closed without merging**